posthog/cdp/templates/attio/template_attio.py

```diff
@@ -1,6 +1,254 @@
+    if (not empty(value)) {
+        userBody.data.values[key] := value
+    }
+}
+
+let personRes := fetch(f'https://api.attio.com/v2/objects/people/records?matching_attribute=email_addresses', {
+    'method': 'PUT',
+    'headers': {
+        'Authorization': f'Bearer {inputs.apiKey}',
+        'Content-Type': 'application/json',
+    },
+    'body': personBody
+})
+
+
+if (personRes.status >= 400) {
+    throw Error(f'Error creating Person from api.attio.com (status {personRes.status}): {personRes.body}')
+}
+
+let userRes := fetch(f'https://api.attio.com/v2/objects/users/records?matching_attribute=user_id', {
+    'method': 'PUT',
+    'headers': {
+        'Authorization': f'Bearer {inputs.apiKey}',
+        'Content-Type': 'application/json',
+    },
+    'body': userBody
+})
+
+if (userRes.status >= 400) {
+    throw Error(f'Error creating User from api.attio.com (status {userRes.status}): {userRes.body}')
+}
+""".strip(),
+    inputs_schema=[
+        {
+            "key": "apiKey",
+            "type": "string",
+            "label": "Access token",
+            "description": "Check out this page to get your API key: https://attio.com/help/reference/integrations-automations/generating-an-api-key",
+            "secret": True,
+            "required": True,
+        },
+        {
+            "key": "email",
+            "type": "string",
+            "label": "Email of the user",
+            "description": "Where to find the email for the user to be created. You can use the filters section to filter out unwanted emails or internal users.",
+            "default": "{person.properties.email}",
+            "secret": False,
+            "required": True,
+        },
+        {
+            "key": "userId",
+            "type": "string",
+            "label": "User ID",
+            "description": "ID of the user in your database. This is a required attribute for the User record in Attio.",
+            "default": "{event.distinct_id}",
+            "secret": False,
+            "required": True,
+        },
+        {
+            "key": "userAttributes",
+            "type": "dictionary",
+            "label": "Additional User attributes",
+            "description": "This object's keys should be the slugs or IDs of the User attributes you wish to update. For information on potential custom attributes, refer to the attribute type docs: https://developers.attio.com/docs/attribute-types",
+            "secret": False,
+            "required": True,
+        },
+        {
+            "key": "personAttributes",
+            "type": "dictionary",
+            "label": "Additional Person attributes",
+            "description": "This object's keys should be the slugs or IDs of the Person attributes you wish to update. For information on potential custom attributes, refer to the attribute type docs: https://developers.attio.com/docs/attribute-types",
+            "secret": False,
+            "required": True,
+        },
```

Comment:
```suggestion
        {
            "key": "userAttributes",
            "type": "dictionary",
            "label": "Additional User attributes",
            "description": "This object's keys should be the slugs or IDs of the User attributes you wish to update. For information on potential custom attributes, refer to the attribute type docs: https://developers.attio.com/docs/attribute-types",
            "secret": False,
            "required": False,
            "default": {}
        },
        {
            "key": "personAttributes",
            "type": "dictionary",
            "label": "Additional Person attributes",
            "description": "This object's keys should be the slugs or IDs of the Person attributes you wish to update. For information on potential custom attributes, refer to the attribute type docs: https://developers.attio.com/docs/attribute-types",
            "secret": False,
            "required": False,
            "default": {}
        },
```

Should probably not be required? They should also have an empty object as a default.